src/java.base/share/native/libjli/java.c

```diff
@@ -387,6 +387,69 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argv */
         } \
     } while (JNI_FALSE)
 
+#define CHECK_EXCEPTION_FAIL() \
+    do { \
+        if ((*env)->ExceptionOccurred(env)) { \
+            (*env)->ExceptionClear(env); \
+            return 0; \
+        } \
+    } while (JNI_FALSE)
+
+
+#define CHECK_EXCEPTION_NULL_FAIL(mainObject) \
+    do { \
+        if ((*env)->ExceptionOccurred(env)) { \
+            (*env)->ExceptionClear(env); \
+            return 0; \
+        } else if (mainObject == NULL) { \
+            return 0; \
+        } \
+    } while (JNI_FALSE)
+
+int
+invokeStaticMainWithArgs(JNIEnv *env, jclass mainClass, jobjectArray mainArgs) {
+    jmethodID mainID = (*env)->GetStaticMethodID(env, mainClass, "main",
+                                  "([Ljava/lang/String;)V");
+    CHECK_EXCEPTION_FAIL();
+    (*env)->CallStaticVoidMethod(env, mainClass, mainID, mainArgs);
+    return 1;
```

Comment:
I think we should add a comment to each of the 4 invokeXXXX functions to say that they return 1 if the main method completes successfully or it throws. Just trying to head off someone asking why there isn't a CHECK_XXX after the CallStaticVoidMethod/CallVoidMethod.

Reply:
Comments added.